scheduler/server/task_scheduler.go

```diff
@@ -164,16 +153,20 @@ func assign(
 				"numAssignments": len(assignments),
 				"tag":            task.Def.Tag,
 			}).Info("Scheduling task")
-		stat.Counter(stats.SchedScheduledTasksCounter).Inc(1)
+		s.stat.Counter(stats.SchedScheduledTasksCounter).Inc(1)
 	}
 	return assignments
 }
 
-func findIdleNodeInGroup(cs *clusterState, groupID string) *nodeState {
-	for _, ns := range cs.nodeGroups[groupID].idle {
-		if ns.suspended() || cs.isOfflined(ns) {
+// findIdleNodeInGroup find a node in the group's idle nodes that is not suspended or offlined (this method will, pick
```

Comment:
this isn't entirely accurate - it's looking through both the idle and busy fields of the nodeGroup

Reply:
nodeGroup is the local copy of clusterState's nodeGroups - this local copy only contains idle nodes 

this local node structure confused me in the first place - hence the bug!  I'll try updating comments, perhaps that will help.  Also recommendations on better variable names are welcome.